fix(docs): Correct multiple typos in free-programming-books-langs.md

BhashkarFulara369 opened the pull request:
This PR addresses several minor typo corrections in `books/free-programming-books-langs.md`:

-   **GraphQL:** Corrected `graphqL` to `GraphQL`.
-   **Python:** Corrected `Al Swigart` to `Al Sweigart`.
-   **Ruby:** Corrected `Casting Spels` to `Casting Spells`.
-   **YAML:** Added a space in `TutorialsPoint` to `Tutorials Point`.

These are minor textual fixes to improve the accuracy and consistency of the list.

Submitted for Hacktoberfest 2025!

eshellman commented on books/free-programming-books-langs.md:
```diff
@@ -849,7 +849,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Fullstack GraphQL](https://github.com/GraphQLCollege/fullstack-graphql) (CC BY-NC-SA)
 * [GraphQL or Bust](https://nordicapis.com/wp-content/uploads/GraphQL-or-Bust-v2.2.pdf) - Nordic APIs (PDF)
-* [Learning graphqL](https://riptutorial.com/Download/graphql.pdf) - Compiled from StackOverflow Documentation (PDF)  (CC BY-SA)
+* [Learning GraphQL](https://riptutorial.com/Download/graphql.pdf) - Compiled from StackOverflow Documentation (PDF)  (CC BY-SA)
```
we use capitalization from source

eshellman commented on books/free-programming-books-langs.md:
```diff
@@ -2082,7 +2082,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [The Python GTK+ 3 Tutorial](http://python-gtk-3-tutorial.readthedocs.org/en/latest/)
 * [The Python Handbook](https://flaviocopes.com/page/python-handbook/) - Flavio Copes (PDF, EPUB, Kindle) *(email address requested)*
-* [The Recursive Book of Recursion](https://inventwithpython.com/recursion/) - Al Swigart (HTML) (3.x)
+* [The Recursive Book of Recursion](https://inventwithpython.com/recursion/) - Al Sweigart (HTML) (3.x)
```
we use titles from the source

eshellman commented on books/free-programming-books-langs.md:
```diff
@@ -1525,7 +1525,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [ANSI Common Lisp Standard (draft version 15.17R, X3J13/94-101R)](https://franz.com/support/documentation/cl-ansi-standard-draft-w-sidebar.pdf) (PDF)
 * [Basic Lisp Techniques](http://franz.com/resources/educational_resources/cooper.book.pdf) - David J. Cooper Jr. (PDF)
-* [Casting Spels in Lisp](http://www.lisperati.com/casting.html)
+* [Casting Spells in Lisp](http://www.lisperati.com/casting.html)
```
we use titles from the source

eshellman commented on books/free-programming-books-langs.md:
```diff
@@ -2710,7 +2710,7 @@ Books on general-purpose programming that don't focus on a specific language are
 ### YAML
 
-* [YAML Tutorial](https://www.tutorialspoint.com/yaml/yaml_tutorial.pdf) - TutorialsPoint (PDF)
+* [YAML Tutorial](https://www.tutorialspoint.com/yaml/yaml_tutorial.pdf) - Tutorials Point (PDF)
```
we use titles from source

BhashkarFulara369 commented:
"Hi @eshellman, I've reviewed your feedback and updated the PR. I've reverted the capitalization for GraphQL, the spelling for 'Casting Spels', and the spacing for 'TutorialsPoint' to match the source titles. The only change remaining is the correction of 'Al Swigart' to 'Al Sweigart' as approved. Please let me know if any further adjustments are needed. Thanks!"

eshellman commented:
Thanks, and welcome to free-programming-books!
